tests/Acceptance/ValidateResult/helper.php

```diff
@@ -19,16 +19,21 @@ function assertNumberTotalErrors(array $reportedJSON, int $numberExpectedErrors)
  * @param array $reportedJSON
  * @param string $givenFileName
  * @param string $errorMessage
+ * @param int $lineNumber
  *
  * @throws RuntimeException if assertion is false
  */
-function assertFileHasErrorMessage(array $reportedJSON, string $givenFileName, string $errorMessage)
+function assertFileHasErrorMessage(array $reportedJSON, string $givenFileName, string $errorMessage, int $lineNumber)
 {
     foreach ($reportedJSON['files'] as $fileName => $fileData) {
         if (basename($fileName) === $givenFileName) {
-            foreach ($fileData['messages'] as $errors) {
-                if ($errors['message'] === $errorMessage) {
-                    return;
+            foreach ($fileData['messages'] as $error) {
+                if ($error['message'] === $errorMessage) {
+                    if ($error['line'] === $lineNumber) {
+                        return;
+                    } else {
+                        throw new RuntimeException(sprintf('Found error message %s at line %d instead of line %d', $errorMessage, $error['line'], $lineNumber));
+                    }
```

Comment:
```suggestion
                    if ($error['line'] === $lineNumber) {
                        return;
                    }

                    throw new RuntimeException(sprintf('Found error message %s at line %d instead of line %d', $errorMessage, $error['line'], $lineNumber));
```